clients/pkg/logentry/stages/extensions.go

```diff
@@ -35,9 +41,46 @@ func NewDocker(logger log.Logger, registerer prometheus.Registerer) (Stage, erro
 	return NewPipeline(logger, stages, nil, registerer)
 }
 
+type cri struct {
+	// bounded buffer for CRI-O Partial logs lines (identified with tag `P` till we reach first `F`)
+	partialLines []string
+	base         *Pipeline
+}
+
+// implement Stage interface
+func (c *cri) Name() string {
+	return "cri"
+}
+
+// implements Stage interface
+func (c *cri) Run(entry chan Entry) chan Entry {
+	entry = c.base.Run(entry)
+
+	in := RunWithSkip(entry, func(e Entry) (Entry, bool) {
+		fmt.Println(e.Extracted)
+		if e.Extracted["flags"] == "P" {
+			if len(c.partialLines) >= MaxPartialLinesSize {
+				c.partialLines = c.partialLines[:0]
+			}
+			c.partialLines = append(c.partialLines, e.Line)
```

Comment:
`RunWithSkip` runs the passed process function in a go routine. Do we need to therefore but any locking around the `partialLines` slice that we're mutating here?

Reply:
Good point. But AFAIK, we don't need any such sync mechanism. Because
1. I think It is using go routine there just because it wants to return the channel directly without waiting for entries to come in.
2. Each pipeline stages in promtail will create separate CRI stage (via `NewCRI`) and all the stages are processed one by one as a pipeline. So I don't think calling CRI pipeline (that has partialline slice) concurrently make any sense.

That's my understanding. But someone can double check. May be @cyriltovena ?